tests/manual/media/FXMediaPlayer/src/fxmediaplayer/media/FXMedia.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package fxmediaplayer.media;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.io.IOException;
+import java.util.stream.Stream;
+import java.nio.file.Files;
+import java.util.function.Predicate;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.nio.file.NoSuchFileException;
+
+/**
+ * Helper class to generate playlist from embeded media files located under
+ * src/fxmediaplayer/media. This is useful to test "file", "jar" and "jrt" protocol.
+ * Copy supported media files under src/fxmediaplayer/media,
+ * compile "FXMediaPlayer", run it and go to "Play List" tab and you should see
+ * your embeded media files.
+ *
+ * File protcol:
+ * cd rt/tests/manual/media/FXMediaPlayer
+ * and run
+ *
+ * JAR protcol:
+ * cd rt/tests/manual/media/FXMediaPlayer
+ * ant
+ * java @../../../../build/run.args -jar dist/FXMediaPlayer.jar
+ *
+ * JRT protocol:
+ * cd rt/tests/manual/media/FXMediaPlayer
+ * ant
+ * [macOS/Linux] jlink --output dist/FXMediaPlayer -p ../../../../build/jmods:dist --add-modules javafx.base,javafx.controls,javafx.media,javafx.graphics,FXMediaPlayer --launcher FXMediaPlayer=FXMediaPlayer/fxmediaplayer.FXMediaPlayer
+ * [macOS/Linux] ./dist/FXMediaPlayer/bin/FXMediaPlayer
+ * [Windows] jlink --output dist/FXMediaPlayer -p ../../../../build/jmods;dist --add-modules javafx.base,javafx.controls,javafx.media,javafx.graphics,FXMediaPlayer --launcher FXMediaPlayer=FXMediaPlayer/fxmediaplayer.FXMediaPlayer
+ * [Windows] dist\FXMediaPlayer\bin\FXMediaPlayer.bat
+ */
```

Comment:
minor: this javadoc will likely not render as expected.  perhaps use <pre> from line 50 to 61 or 65.
also, spelling "protcol" on line 50

Reply:
Also minor: you can shorten the command lines by removing `javafx.base` and `javafx.graphics` from the list of added modules, since `javafx.controls` transiently requires them.

Reply:
Spelling fixed and javafx.base and javafx.graphics is removed. I do not think that we need to worry about javadoc, since we do not generate it for this app.

Reply:
This is just a suggestion, really, especially if it is in test code.
However, Eclipse IDE shows javadoc (something I routinely use, the javadoc tab is always open):

<img width="945" alt="Screen Shot 2022-10-12 at 15 55 55" src="https://user-images.githubusercontent.com/107069028/195462859-1174e2ca-53cd-4f02-9ff3-bd989ccd2cdd.png">

Reply:
Fixed.